### DISCUSSION TITLE
How to filter Work Item Type in Generate release note task

### DISCUSSION THREAD

**Discussion by khopithan** [rfennell/AzurePipelines#1349] [Q&A] [answered]
Hi @rfennell,

Hope you are doing good. I have a requirement that I need to list "Bug, Activity & User story" work item types only in the release note. How can I filter that. Thanks in advance.

Regards,
Khopi

**Comment by rfennell** [comment, marked as answer]
You can just use a filter in the wi loop, something like

```
{{#forEach this.workItems}}
{{#if isFirst}}### WorkItems of type 'Bug'{{/if}}
{{#if (eq (lookup this.fields 'System.WorkItemType') 'Bug')}}
   - **{{this.id}}** : {{lookup this.fields 'System.Title'}}
{{/if}}
{{/forEach}} 
```

The tasks ships with all these [handlebars helpers](https://github.com/helpers/handlebars-helpers#comparison) that should let you build the more complex logic you require.

**Reply by khopithan** [reply]
Thank you....like wise I have to write separate loops for other Work Items as well.

**Reply by rfennell** [reply]
You have plenty of option, depends what you need in the output. 

You could have completely separate loops/if constructs

Or one loop with multiple if blocks you ant different output per WIT

```
{{#forEach this.workItems}}
{{#if isFirst}}### WorkItems{{/if}}
{{#if (eq (lookup this.fields 'System.WorkItemType') 'Bug')}}
   - **{{this.id}}** : {{lookup this.fields 'System.Title'}}
{{/if}}
{{#if (eq (lookup this.fields 'System.WorkItemType') 'Activity')}}
   - **{{this.id}}** : {{lookup this.fields 'System.Title'}}
{{/if}}
{{/forEach}} 
```

If you need the same fields then a combined `if`  and `or` blocks

```
{{#forEach this.workItems}}
{{#if isFirst}}### WorkItems of type 'Bug'{{/if}}
{{#if or ((eq (lookup this.fields 'System.WorkItemType') 'Bug') (eq (lookup this.fields 'System.WorkItemType') 'Activity'))}}
   - **{{this.id}}** : {{lookup this.fields 'System.Title'}}
{{/if}}
{{/forEach}} 
```

But as I said there might be a more elegant way with one of the [helpers](https://github.com/helpers/handlebars-helpers#comparison)

**Reply by khopithan** [reply]
Thank you so much...😊😊
